python/lsst/cp/verify/verifyCalib.py

```diff
@@ -0,0 +1,193 @@
+                               defaultTemplates={}):
+    inputCalib = cT.Input(
+        name="calib",
+        doc="Input calib to calculate statistics for.",
+        storageClass="IsrCalib",
+        dimensions=["instrument", "detector"],
+        isCalibration=True
+    )
+
+    outputStats = cT.Output(
+        name="calibStats",
+        doc="Output statistics from cp_verify.",
+        storageClass="StructuredDataDict",
+        dimensions=["instrument", "detector"],
+    )
+
+
+class CpVerifyCalibConfig(pipeBase.PipelineTaskConfig,
+                          pipelineConnections=CpVerifyCalibConnections):
+    """Configuration parameters for CpVerifyCalibTask.
+    """
+    # Statistics options.
+    useReadNoise = pexConfig.Field(
+        dtype=bool,
+        doc="Compare sigma against read noise?",
+        default=True,
+    )
+    numSigmaClip = pexConfig.Field(
+        dtype=float,
+        doc="Rejection threshold (sigma) for statistics clipping.",
+        default=5.0,
+    )
+    clipMaxIter = pexConfig.Field(
+        dtype=int,
+        doc="Max number of clipping iterations to apply.",
+        default=3,
+    )
+
+    # Keywords and statistics to measure from different sources.
+    calibStatKeywords = pexConfig.DictField(
+        keytype=str,
+        itemtype=str,
+        doc="Calib statistics to run.",
+        default={},
+    )
+
+
+class CpVerifyCalibTask(pipeBase.PipelineTask, pipeBase.CmdLineTask):
+    """Main statistic measurement and validation class.
+
+    This operates on a generic calibration, and is designed to be
+    subclassed so specific calibrations can apply their own validation
+    methods.
+    """
+
+    ConfigClass = CpVerifyCalibConfig
+    _DefaultName = 'cpVerifyCalib'
+
+    def run(self, inputCalib):
+        """Calculate quality statistics and verify they meet the requirements
+        for a calibration.
+
+        Parameters
+        ----------
+        inputCalib : `lsst.ip.isr.IsrCalib`
+            The calibration to be measured.
+
+        Returns
+        -------
+        result : `lsst.pipe.base.Struct`
+            Result struct with components:
+            - ``outputStats`` : `dict`
+                The output measured statistics.
+
+        Notes
+        -----
+        The outputStats should have a yaml representation of the form
+
+        AMP:
+          Amp1:
+            STAT: value
+            STAT2: value2
+          Amp2:
+          Amp3:
+        DET:
+          STAT: value
+          STAT2: value
+        CATALOG:
+          STAT: value
+          STAT2: value
+        VERIFY:
+          DET:
+            TEST: boolean
+          CATALOG:
+            TEST: boolean
+          AMP:
+            Amp1:
+              TEST: boolean
+              TEST2: boolean
+            Amp2:
+            Amp3:
+        SUCCESS: boolean
+
+        """
+        outputStats = {}
+
+        outputStats['DETECTOR'] = self.detectorStatistics(inputCalib)
```

Comment:
I don't see how this could create a dict that resolves to the yaml described in the docstring.  Roots would be `DETECTOR`, `VERIFY`, `SUCCESS`, and not `AMP`, `DET`, `CATALOG`.

Reply:
Also below there's something about `N_VALID`?

Reply:
This is a copy from the `CpVerifyStatsTask` block, and I forgot to clean it up to match reality.  I've removed the entries that do not exist, and renamed `DETECTOR` to `DET` to match how `verifyStats` names things.  `N_VALID` is one of the `STAT` entries, and I've updated the text to note that.